lib/exercism/user_statistics.rb

```diff
@@ -0,0 +1,31 @@
+class UserStatistics
+  def self.of(username)
+    by_username(username)
+  end
+
+  def self.by_username(username)
+    user = User.find_by(username: username)
+    { user: user.public_user_attributes,
+      statistics: user_problem_statistcs(user) }
+  end
+
+  def self.problem_total(track)
+    track.problems.count
+  end
+
+  def self.problem_completion(user, track)
+    user.exercises.where("LOWER(language) = ?", track.language.downcase).map do |exercise|
```

Comment:
We're using postgresql, which is case insensitive, so we don't need to do the `LOWER` thing, or the downcase. On the other hand, `track.language` is the name of the language and `UserExercise.language` is the track ID, so this needs to be `user.exercises.where(language: track.id).map`